SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java

```diff
@@ -1331,20 +1349,21 @@ private final boolean getBooleanOption(final String option) {
 	}
 
 	public boolean isTerminationRequested() {
-		return mCancel.isTerminationRequested();
+		return mResourceLimit.isTerminationRequested();
 	}
 
 	/**
 	 * Use with caution.
 	 */
 	public void setTerminationRequest(final TerminationRequest request) {
-		mCancel = new TimeoutHandler(request);
+		mTimeout = new TimeoutHandler(request);
+		mResourceLimit = new ResourceLimit(mTimeout);
 	}
 
 	/**
 	 * Use with caution.
 	 */
 	public TerminationRequest getTerminationRequest() {
-		return mCancel.getTerminationRequest();
+		return mResourceLimit;
```

Comment:
I think it should return mCancel, to match setTerminationRequest (and the previous behavior).

Reply:
I check the setter and getter to much the previous behaviour.